lib/presto/client/statement_client.rb

```diff
@@ -40,6 +40,15 @@ def initialize(faraday, query, options, next_uri=nil)
         @models = Models
       end
 
+      @plan_timeout = options[:plan_timeout]
+      @query_timeout = options[:query_timeout]
+
+      if @plan_timeout || @query_timeout
+        # this is set before the first call of faraday_get_with_retry so that
+        # resuming StatementClient with next_uri is also under timeout control.
+        @query_submitted_at = Time.now
```

Comment:
Better use `Process.clock_gettime(Process::CLOCK_MONOTONIC)`?

Reply:
CLOCK_MONOTONIC sounds more appropriate.
I created another PR for existent use of Time.now: https://github.com/treasure-data/presto-client-ruby/pull/50

Reply:
ok, CLOCK_MONOTONIC doesn't work on Circle CI...

Reply:
disabled Circle CI. it's just broken.